### PR TITLE
[WebPubSub] Make hub methods virtual.

### PR DESCRIPTION
**Pull request by JialinXin**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/WebPubSubHub.cs [hunk 1]
```diff
@@ -18,15 +18,21 @@ public abstract class WebPubSubHub
         /// <param name="request"><see cref="ConnectedEventRequest"/> to get client connect request information.</param>
         /// <param name="cancellationToken">The <see cref="CancellationToken"/> used to cancel the request.</param>
         /// <returns>Build a valid <see cref="WebPubSubEventResponse"/> with ConnectEventRequest.CreateResponse().</returns>
-        public abstract ValueTask<WebPubSubEventResponse> OnConnectAsync(ConnectEventRequest request, CancellationToken cancellationToken);
+        public virtual ValueTask<WebPubSubEventResponse> OnConnectAsync(ConnectEventRequest request, CancellationToken cancellationToken)
+        {
+            return default;
```
Comment (vicancy) [inline sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/WebPubSubHub.cs:23]: How about by default returning a `WebPubSubEventResponse.Success`

Comment (JialinXin) [reply sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/WebPubSubHub.cs:23]: `default` is body null and status code 200, same as `Success`. Do you suggest to specifically define the `Success` for users to easy understand?

Comment (JialinXin) [reply sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/WebPubSubHub.cs:23]: We have the `CreateResponse()` methods for user defined parameters in normal cases, and in null case, user can return `base.OnConnectAsync(request, ctk)`. Not much need to define `Success` anymore.

File: sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/Internal/ServiceRequestHandlerAdapter.cs [hunk 2]
```diff
@@ -154,7 +141,12 @@ public async Task HandleRequest(HttpContext context)
             {
                 context.Response.StatusCode = StatusCodes.Status401Unauthorized;
                 await context.Response.WriteAsync(ex.Message).ConfigureAwait(false);
-                return;
+            }
+            catch (Exception ex)
+            {
+                // logging to service.
+                context.Response.StatusCode = StatusCodes.Status500InternalServerError;
```
Comment (vicancy) [inline sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/Internal/ServiceRequestHandlerAdapter.cs:148]: also leverage ILogger and print some error logs in the SDK?

**Review by vicancy — APPROVED** [review]
Overall it looks pretty good to me

File: sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/Internal/ServiceRequestHandlerAdapter.cs [hunk 3]
```diff
@@ -17,21 +18,24 @@ internal class ServiceRequestHandlerAdapter
     {
         private readonly WebPubSubOptions _options;
         private readonly IServiceProvider _provider;
+        private readonly ILogger _logger;
 
         // <hubName, HubImpl>
         private readonly Dictionary<string, WebPubSubHub> _hubRegistry = new(StringComparer.OrdinalIgnoreCase);
 
-        public ServiceRequestHandlerAdapter(IServiceProvider provider, IOptions<WebPubSubOptions> options)
+        public ServiceRequestHandlerAdapter(IServiceProvider provider, IOptions<WebPubSubOptions> options, ILoggerFactory loggerFactory)
```
Comment (vicancy) [inline sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/Internal/ServiceRequestHandlerAdapter.cs:26]: you can directly use ILogger<ServiceRequestHandlerAdapter>

File: sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/Internal/ServiceRequestHandlerAdapter.cs [hunk 4]
```diff
@@ -17,21 +18,24 @@ internal class ServiceRequestHandlerAdapter
...
         {
             _provider = provider;
             _options = options.Value;
+            _logger = loggerFactory?.CreateLogger<ServiceRequestHandlerAdapter>();
         }
 
         // for tests.
-        internal ServiceRequestHandlerAdapter(WebPubSubOptions options, WebPubSubHub hub)
+        internal ServiceRequestHandlerAdapter(WebPubSubOptions options, WebPubSubHub hub, ILogger logger)
```
Comment (vicancy) [inline sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/Internal/ServiceRequestHandlerAdapter.cs:34]: Can we get rid of this ctor?

**Pull request merged**